code/modules/antagonists/wizard/equipment/artefact.dm

```diff
@@ -299,78 +299,81 @@
+	desc = "Calls a cloud to come pick you up and drop you at a random location on the station."
 	icon = 'icons/obj/wizard.dmi'
 	icon_state = "whistle"
-	var/on_cooldown = 0 //0: usable, 1: in use, 2: on cooldown
-	var/mob/living/carbon/last_user
 
-/obj/item/warpwhistle/proc/interrupted(mob/living/carbon/user)
-	if(!user || QDELETED(src) || user.notransform)
-		on_cooldown = FALSE
-		return TRUE
-	return FALSE
+	/// Cooldown between whistle uses.
+	COOLDOWN_DECLARE(whistle_cooldown)
+	/// Person using the warp whistle
+	var/mob/living/whistler
 
-/obj/item/warpwhistle/proc/end_effect(mob/living/carbon/user)
-	user.invisibility = initial(user.invisibility)
-	user.status_flags &= ~GODMODE
-	REMOVE_TRAIT(user, TRAIT_IMMOBILIZED, WARPWHISTLE_TRAIT)
-
-
-/obj/item/warpwhistle/attack_self(mob/living/carbon/user)
-	if(!istype(user) || on_cooldown)
-		return
-	on_cooldown = TRUE
-	last_user = user
-	var/turf/T = get_turf(user)
-	playsound(T,'sound/magic/warpwhistle.ogg', 200, TRUE)
-	ADD_TRAIT(user, TRAIT_IMMOBILIZED, WARPWHISTLE_TRAIT)
-	new /obj/effect/temp_visual/tornado(T)
-	sleep(20)
-	if(interrupted(user))
-		REMOVE_TRAIT(user, TRAIT_IMMOBILIZED, WARPWHISTLE_TRAIT)
+/obj/item/warp_whistle/attack_self(mob/user)
+	if(!COOLDOWN_FINISHED(src, whistle_cooldown))
+		to_chat(user, span_warning("[src] is still on cooldown!"))
 		return
-	user.invisibility = INVISIBILITY_MAXIMUM
-	user.status_flags |= GODMODE
-	sleep(20)
-	if(interrupted(user))
-		end_effect(user)
-		return
-	var/breakout = 0
-	while(breakout < 50)
-		var/turf/potential_T = find_safe_turf()
-		if(T.z != potential_T.z || abs(get_dist_euclidian(potential_T,T)) > 50 - breakout)
-			do_teleport(user, potential_T, channel = TELEPORT_CHANNEL_MAGIC)
-			T = potential_T
-			break
-		breakout += 1
-	new /obj/effect/temp_visual/tornado(T)
-	sleep(20)
-	end_effect(user)
-	if(interrupted(user))
+	if(whistler)
+		to_chat(user, span_warning("[src] is already warping."))
 		return
-	on_cooldown = 2
-	addtimer(VARSET_CALLBACK(src, on_cooldown, 0), 4 SECONDS)
 
-/obj/item/warpwhistle/Destroy()
-	if(on_cooldown == 1 && last_user) //Flute got dunked somewhere in the teleport
-		end_effect(last_user)
-	return ..()
+	whistler = user
+	var/turf/current_turf = get_turf(user)
+	var/turf/spawn_location = locate(user.x + pick(-7, 7), user.y, user.z)
+	playsound(current_turf,'sound/magic/warpwhistle.ogg', 200, TRUE)
+	new /obj/effect/temp_visual/teleporting_tornado(spawn_location, src)
+	COOLDOWN_START(src, whistle_cooldown, 4 SECONDS)
 
-/obj/effect/temp_visual/tornado
-	icon = 'icons/obj/wizard.dmi'
-	icon_state = "tornado"
+///Teleporting tornado, spawned by warp whistle, teleports the user if they manage to pick them up.
+/obj/effect/temp_visual/teleporting_tornado
 	name = "tornado"
 	desc = "This thing sucks!"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "tornado"
 	layer = FLY_LAYER
 	plane = ABOVE_GAME_PLANE
-	randomdir = 0
-	duration = 40
-	pixel_x = 500
+	randomdir = FALSE
+	duration = 10 SECONDS
```

Comment:
will it still look right?

Reply:
It’ll fly away and lower alpha so it should be fine, I don’t want to cut it too close in case it catches the summoner later